src/operator/optimizer_op.cc

```diff
@@ -105,9 +106,35 @@ It updates the weights using::
 .add_argument("var", "NDArray-or-Symbol", "Moving variance")
 .add_arguments(AdamParam::__FIELDS__());
 
+
 NNVM_REGISTER_OP(rmsprop_update)
-.describe(R"code(Update function for RMSProp optimizer. The RMSProp code follows the version in
-http://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf Tieleman & Hinton, 2012.
+.describe(R"code(Update function for `RMSProp` optimizer.
+
+The `RMSprop` is a form of stochastic gradient descent where the gradients are
+divided by a running average of their recent magnitude. `RMSprop` was developped
+from the need to resolve `Adagrad`'s radically diminishing learning rates.
```

Comment:
This could be clearer. 
Instead of "`RMSprop` was developed from the need to resolve `Adagrad`'s radically diminishing learning rates."

Please write:

"`RMSProp` is similar to AdaGrad, a popular variant of SGD which adaptively tunes the learning rate of each parameter. AdaGrad lowers the learning rate for each parameter monotonically over the course of training. While this is analytically motivated for convex optimizations, it may not be ideal for non-convex problems. RMSProp deals with this heuristically by allowing the learning rates to rebound as the denominator decays over time."